### DISCUSSION TITLE
How to get the current panel anywhere in the application?

### DISCUSSION THREAD

**Discussion by discoverlance-com** [filamentphp/filament#12274] [Help]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I am using filamentphp with multiple admin panels. In this case, let's assume it's user and admin. I have added social login but there's a problem in my social login redirect page, when there's an error with the login state and I want to redirect to the login page, for some reason, it's always redirecting to the admin login page even if the user logged in from the user admin panel. How can I ensure it maintains the panel where the login was initiated and when I redirect to the login page from the callback url, it will go to the login page of the panel where the login was initiated?

My callback page redirect:
```php
    // I tried this which uses the getLoginUrl function but it always redirects to the admin login url
      if (! $this->isValidProvider($provider)) {
            redirect(Filament::getLoginUrl());

            return;
     }
    // I tried this which uses the currentPanel function but it's not getting the current panel
      if (! $this->isValidProvider($provider)) {
            redirect(Filament::getCurrentPanel()->getLoginUrl());

            return;
       }
```

**Comment by ariesdg** [comment]
were you able to resolve this?

**Reply by discoverlance-com** [reply]
> were you able to resolve this?

I could not make it work so in the end, I used different pages to handle redirects for each admin panel. So I did not need to check for the current panel as should the user visit, say `/admin/callback`, I know it's for the admin panel so I write code for the admin panel.

**Comment by juliangums** [comment]
@ariesdg @discoverlance-com 
This plugin already solved this for you. 
https://v2.filamentphp.com/plugins/socialite

Most drivers will work by default, for some drivers you can try this branch (might already be merged into main by the time you read this):
https://github.com/DutchCodingCompany/filament-socialite/tree/improve-routing
You can find out more here: https://github.com/DutchCodingCompany/filament-socialite/issues/58#issuecomment-2186766910
